File: src/ProjectR/content/scripts/spells/HandOfDeath.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Hand of Death"
end

function GetDescription()
  return "Attempt to kill all enemies.\nHigh damage modifier."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 1.88
end

function GetDelay()
  return .4
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { DRK }
end

function SpellEffect()
  d:ApplyDebuff(DTH, 100)
  local damage = (9 * aMD * (aDRK/100) - 1.5 * dMR) * (100/dDRK)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/HandOfDeath.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class HandOfDeath : SpellScriptBase
    {
        public override string Name { get { return "Hand Of Death"; } }
        public override string Description { get { return "Attempt to kill all enemies.\nHigh damage modifier."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.DRK, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 188; } }
        public override double Delay { get { return .4; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (9 * cMD * (cDRK / 100) - 1.5 * tMR) * (100 / tDRK);
            DealDamage(damage);
            TryToApplyDebuff(DebuffResistance.DTH, 100);
        }
    }
}